Source/JavaScriptCore/dfg/DFGSpeculativeJIT.cpp

```diff
@@ -9579,7 +9579,21 @@ void SpeculativeJIT::compileArrayIndexOf(Node* node)
 
     case StringUse: {
         ASSERT(node->arrayMode().type() == Array::Contiguous);
+#if USE(JSVALUE32_64)
+        SpeculateCellOperand searchElement(this, searchElementEdge);
+
+        GPRReg searchElementGPR = searchElement.gpr();
+
+        speculateString(searchElementEdge, searchElementGPR);
+
+        flushRegisters();
+
+        callOperation(operationArrayIndexOfString, lengthGPR, LinkableConstant::globalObject(*this, node), storageGPR, searchElementGPR, indexGPR);
+
+        strictInt32Result(lengthGPR, node);
 
+        return;
+#else
```

Comment:
It still fails on 32-bit and It is hard to debug because I do not have a 32-bit machine so falls back to JIT Operation.